File: fsms-backend-api/src/main/java/com/fuelstation/managmentapi/common/domain/Actor.java
package com.fuelstation.managmentapi.common.domain;

import com.fuelstation.managmentapi.authentication.domain.UserRole;

public record Actor(Long id, ActorType type, UserRole role) {

    public static Actor system() {
        return new Actor(null, ActorType.SYSTEM, null);
    }

    public static Actor user(long userId, UserRole role) {
        return new Actor(userId, ActorType.USER, role);
    }

    public boolean isSystem() {
        return type == ActorType.SYSTEM;
    }

    public boolean isUser() {
        return type == ActorType.USER;
    }

    public boolean isAdmin() { return isUser() && role == UserRole.ADMINISTRATOR; }

    public boolean isManager() { return isUser() && role == UserRole.MANAGER; }

}

File: fsms-client-sdk/projects/fsms-web-api/src/lib/core/actor.ts
export enum ActorType {
    User = "USER",
    System = "SYSTEM"
}

export class Actor {

    constructor(public id: number, public type: ActorType) {}

    system(): boolean {
        return this.type === ActorType.System;
    }

    user(): boolean {
        return this.type === ActorType.User;
    }

}